cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/ReplicationService.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        // Create a background priority thread to so we don't block the process's main thread.
+        HandlerThread thread = new HandlerThread("ServiceStartArguments",
+            android.os.Process.THREAD_PRIORITY_BACKGROUND);
+        thread.start();
+
+        // Get the HandlerThread's Looper and use it for our Handler.
+        Looper serviceLooper = thread.getLooper();
+        mServiceHandler = getHandler(serviceLooper);
+
+        Replicator[] replicators = getReplicators(getApplicationContext());
+
+        if (replicators != null && replicators.length > 0) {
+            if (mReplicationPolicyManager == null) {
+                mReplicationPolicyManager = new ReplicationPolicyManager() {
+                    @Override
+                    public void start() {
+                    }
+
+                    @Override
+                    public void stop() {
+                    }
+                };
+            }
+            mReplicationPolicyManager.addReplicators(replicators);
+            mReplicationPolicyManager.setReplicationsCompletedListener(this);
+            mReplicationPolicyManager.start();
+        } else {
+            throw new RuntimeException(
+                    "No replications setup. Please return Replicators from getReplicators()");
+        }
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        // Extract the command from the given Intent and pass it to our handler to process the
+        // command on a separate thread.
+        if (intent != null && intent.hasExtra(EXTRA_COMMAND)) {
+            Message msg = mServiceHandler.obtainMessage();
+            msg.arg1 = startId;
+            msg.arg2 = intent.getIntExtra(EXTRA_COMMAND, COMMAND_STOP_REPLICATION);
+            Bundle bundle = new Bundle();
+            bundle.putParcelable(EXTRA_INTENT, intent);
+            msg.setData(bundle);
+            mServiceHandler.sendMessage(msg);
+        }
+
+        return START_REDELIVER_INTENT;
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mBinder;
+    }
+
+    @Override
+    public boolean onUnbind(Intent intent) {
+        // Ensure onRebind is called when new clients bind to the service.
+        return true;
+    }
+
+    @Override
+    public void onRebind(Intent intent) {
+        super.onRebind(intent);
+    }
+
+    /**
+     * Set the {@link ReplicationPolicyManager} to be used by this ReplicationService.
+     * @param replicationPolicyManager
+     */
+    public void setReplicationPolicyManager(ReplicationPolicyManager replicationPolicyManager) {
+        mReplicationPolicyManager = replicationPolicyManager;
+    }
+
+    /**
+     * Start the set of replications specified in the set of {@link Replicator}s returned by
+     * {@link #getReplicators(Context)}.
+     */
+    protected void startReplications() {
```

Comment:
Does this always run on the same thread? If not, it would appear we could lose the reference to a held wifi lock.